src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -68,14 +68,16 @@ public T Value
 
     [DataContract(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/Arrays")]
 #if USE_REFEMIT
-    public struct KeyValue<K, V>
+    public class KeyValue<K, V>
 #else
-    internal struct KeyValue<K, V>
```

Comment:
Good point. I've reverted the change of KeyValue back to struct to avoid potential GC impact. I tried a few alternatives to this and realized that Activator.CreateInstance on a struct type uses a special IL for struct and so just calling Activator.CreateInstance will work. I get the numbers for this change and they are as good as the previous change for dictionary scenario.